File: js/tooltip.js
const submitTooltip = document.querySelector(".tooltip");
const newPromptTooltip = document.querySelector(".tooltip2");
const promptOfTheDayTooltip = document.querySelector(".tooltip3");
const newPoemTooltip = document.querySelector(".tooltip4");
const infoTooltip = document.querySelector(".tooltip5");

toggleInfoButton.addEventListener("mousemove", function(event) {
  const x = event.clientX;
  const y = event.clientY;
  infoTooltip.style.top = y - 30 + "px";
  infoTooltip.style.left = x - 43 + "px";
});

if (currentPage === 'index.html') {
  submitImg.addEventListener("mousemove", function(event) {
    const x = event.clientX;
    const y = event.clientY;
    submitTooltip.style.top = y - 45 + "px";
    submitTooltip.style.left = x - 50 + "px";
  });

  newPrompt.addEventListener("mousemove", function(event) {
    const x = event.clientX;
    const y = event.clientY;
    newPromptTooltip.style.top = y - 40 + "px";
    newPromptTooltip.style.left = x - 40 + "px";
  });

  poemOfTheDay.addEventListener("mousemove", function(event) {
    const x = event.clientX;
    const y = event.clientY;
    promptOfTheDayTooltip.style.top = y - 40 + "px";
    promptOfTheDayTooltip.style.left = x - 55 + "px";
  });

  newPoem.addEventListener("mousemove", function(event) {
    const x = event.clientX;
    const y = event.clientY;
    newPoemTooltip.style.top = y - 40 + "px";
    newPoemTooltip.style.left = x - 30 + "px";
  });
}







/*const tooltip = document.getElementById('tooltip4');
const promptText = document.getElementById('prompt-text');

// Update tooltip text
promptText.addEventListener('mouseenter', () => {
  tooltip.textContent = 'your writing prompt'; // Change the tooltip text dynamically here
});

promptText.addEventListener('mouseleave', () => {
  tooltip.style.display = 'none'; // Hide the tooltip when the user leaves the text
});*/

File: rust/tooltip.rs
use stdweb::web::{
    document,
    event::{MouseMoveEvent, IEventTarget},
    IHtmlElement,
};

fn main() {
    let submit_tooltip = document().query_selector(".tooltip").unwrap();
    let new_prompt_tooltip = document().query_selector(".tooltip2").unwrap();
    let prompt_of_the_day_tooltip = document().query_selector(".tooltip3").unwrap();
    let new_poem_tooltip = document().query_selector(".tooltip4").unwrap();
    let info_tooltip = document().query_selector(".tooltip5").unwrap();

    let submit_img = document().query_selector("#submitImg").unwrap();
    submit_img.add_event_listener(|event: MouseMoveEvent| {
        let x = event.client_x();
        let y = event.client_y();
        submit_tooltip.style().set_property("top", &(y - 45).to_string() + "px").unwrap();
        submit_tooltip.style().set_property("left", &(x - 50).to_string() + "px").unwrap();
    });

    let new_prompt = document().query_selector("#newPrompt").unwrap();
    new_prompt.add_event_listener(|event: MouseMoveEvent| {
        let x = event.client_x();
        let y = event.client_y();
        new_prompt_tooltip.style().set_property("top", &(y - 40).to_string() + "px").unwrap();
        new_prompt_tooltip.style().set_property("left", &(x - 40).to_string() + "px").unwrap();
    });

    let poem_of_the_day = document().query_selector("#poemOfTheDay").unwrap();
    poem_of_the_day.add_event_listener(|event: MouseMoveEvent| {
        let x = event.client_x();
        let y = event.client_y();
        prompt_of_the_day_tooltip.style().set_property("top", &(y - 40).to_string() + "px").unwrap();
        prompt_of_the_day_tooltip.style().set_property("left", &(x - 55).to_string() + "px").unwrap();
    });

    let new_poem = document().query_selector("#newPoem").unwrap();
    new_poem.add_event_listener(|event: MouseMoveEvent| {
        let x = event.client_x();
        let y = event.client_y();
        new_poem_tooltip.style().set_property("top", &(y - 40).to_string() + "px").unwrap();
        new_poem_tooltip.style().set_property("left", &(x - 30).to_string() + "px").unwrap();
    });

    let toggle_info_button = document().query_selector("#toggleInfoButton").unwrap();
    toggle_info_button.add_event_listener(|event: MouseMoveEvent| {
        let x = event.client_x();
        let y = event.client_y();
        info_tooltip.style().set_property("top", &(y - 30).to_string() + "px").unwrap();
        info_tooltip.style().set_property("left", &(x - 43).to_string() + "px").unwrap();
    });
}
